File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/snap/Snap.java
package com.mapbox.services.android.navigation.v5.snap;

import android.location.Location;

import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;

public abstract class Snap {

  public abstract Location getSnappedLocation(Location location, RouteProgress routeProgress);
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/snap/Snap.kt
package com.mapbox.services.android.navigation.v5.snap

import android.location.Location
import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress

abstract class Snap {

    abstract fun getSnappedLocation(location: Location, routeProgress: RouteProgress): Location
}
